### PR TITLE
fix: default status code colors and sorting logic

### PR DESCRIPTION
**Pull request by filipgutica**
# Summary

https://konghq.atlassian.net/browse/MA-2486
https://konghq.atlassian.net/browse/MA-2832

  - apply status code and status code group colors by default
  - apply status code sorting by default
  - apply ____OTHER____ legend and tooltip sorting by default

#### Resources

- [Creating a package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md)

### CODE REVIEW

File: packages/analytics/analytics-chart/src/utils/customColors.ts [hunk 1]
```diff
@@ -67,3 +69,20 @@ export const lookupStatusCodeCategoryColor = (label: string) => {
 
   return { solid: color, light: color }
 }
+
+const statusCodes = Array.from({ length: 500 }, (_, index) => 100 + index)
+
+export const defaultStatusCodeColorPallette: AnalyticsChartColors = {
```
Comment (mihai-peteu) [inline packages/analytics/analytics-chart/src/utils/customColors.ts:75]: nit: can we just make it `defaultStatusCodeColors`? 

I know things autofill in VS Code nowadays, but I still prefer shorter names; plus I have a 25% chance of spelling "pallette" correctly. And i'm pretty sure that's misspelled above too 🤣

File: packages/analytics/analytics-chart/sandbox/pages/TimeSeriesChartDemo.vue [hunk 2]
```diff
@@ -357,7 +357,7 @@ const analyticsChartOptions = computed<AnalyticsChartOptions>(() => {
   return {
     type: chartType.value,
     stacked: stackToggle.value,
-    chartDatasetColors: colorPalette.value,
+    // chartDatasetColors: colorPalette.value,
```
Comment (mihai-peteu) [inline packages/analytics/analytics-chart/sandbox/pages/TimeSeriesChartDemo.vue]: just make sure to remove this commented out line, and the one above. I assume the fallbacks are now working as expected.

File: packages/analytics/analytics-chart/sandbox/pages/TimeSeriesChartDemo.vue [hunk 3]
```diff
@@ -296,7 +296,7 @@ const metricItems = [{
 
 // Short labels
 const statusCodeLabels = [
-  '200', '300', '400', '500',
+  '200', '300', '400', '500', '5XX',
```
Comment (mihai-peteu) [inline packages/analytics/analytics-chart/sandbox/pages/TimeSeriesChartDemo.vue:299]: Why only `5XX` here, and not other grouped status codes as well?

**Review by mihai-peteu — CHANGES_REQUESTED** [review]
Just a few nits

File: packages/analytics/analytics-chart/sandbox/pages/TimeSeriesChartDemo.vue (hunk 3 above)
Comment (filipgutica) [reply packages/analytics/analytics-chart/sandbox/pages/TimeSeriesChartDemo.vue:299]: This is just the sandbox...  just testing stuff out

File: packages/analytics/analytics-chart/src/components/chart-types/TimeSeriesChart.vue [hunk 4]
```diff
@@ -133,11 +132,6 @@ const props = defineProps({
     required: false,
     default: '',
   },
-  datasetColors: {
```
Comment (filipgutica) [inline packages/analytics/analytics-chart/src/components/chart-types/TimeSeriesChart.vue:136]: This was redundant

File: packages/analytics/analytics-chart/src/components/chart-types/TimeSeriesChart.vue [hunk 5]
```diff
@@ -197,23 +191,6 @@ const { options } = composables.useLinechartOptions({
...
-      } else if (e.rawDimension in props.datasetColors) {
-        e.backgroundColor = props.datasetColors[e.rawDimension]
-        e.borderColor = darkenColor(e.backgroundColor, 50)
-      }
-      e.fill = props.fill
-      return e
-    }),
-  }
-})
-
```
Comment (filipgutica) [inline packages/analytics/analytics-chart/src/components/chart-types/TimeSeriesChart.vue:216]: This was all redundant as well, props are all reactive by default, my mistake from the initial implementation.

Comment (filipgutica) [reply packages/analytics/analytics-chart/src/components/chart-types/TimeSeriesChart.vue:216]: Verified in the sandbox, colors and fill still update accordingly when the prop is updated.

File: packages/analytics/analytics-chart/sandbox/pages/TimeSeriesChartDemo.vue (hunk 3 above)
Comment (mihai-peteu) [reply packages/analytics/analytics-chart/sandbox/pages/TimeSeriesChartDemo.vue:299]: oh my bad. Disregard :)

**Review by mihai-peteu — APPROVED** [review]

**Review by mihai-peteu — APPROVED** [review]

**Pull request merged**